worlds/factorio/__init__.py

```diff
@@ -545,7 +543,8 @@ def __init__(self, player: int, name: str, address: int, parent: Region):
 
         self.ingredients = {Factorio.ordered_science_packs[self.complexity]: 1}
         for complexity in range(self.complexity):
-            if parent.multiworld.tech_cost_mix[self.player] > parent.multiworld.random.randint(0, 99):
+            if (parent.multiworld.worlds[self.player].options.tech_cost_mix >
+                    parent.multiworld.random.randint(0, 99)):
```

Comment:
Could do this too, I suppose
```suggestion
                    parent.multiworld.worlds[self.player].random.randint(0, 99)):
```